src/verify_cert.rs

```diff
@@ -705,8 +708,10 @@ mod tests {
 
     #[test]
     fn eku_key_purpose_id() {
```

Comment:
Since the issue that motivated this change was missing API surface I think it would be helpful to land an integration test under `tests/` outside of `src/verify_cert.rs` that uses the public API to demonstrate the feature working as desired. That way we can't accidentally backslide making part of this crate-internal.